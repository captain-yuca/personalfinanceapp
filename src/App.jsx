import React from 'react'
import { BrowserRouter } from 'react-router-dom'
import { ThemeProvider } from 'styled-components'

import { theme } from './config/theme'
import Router from './Router'
import { GlobalStyles } from './styles/GlobalStyles'
import { ReactLogo, StyledApp } from './styles/App.styles'

function App() {
  return (
    <ThemeProvider theme={theme}>
      <GlobalStyles />
      <StyledApp>
        <ReactLogo />
        <BrowserRouter>
          <Router />
        </BrowserRouter>
      </StyledApp>
    </ThemeProvider>
  )
}

export default App
