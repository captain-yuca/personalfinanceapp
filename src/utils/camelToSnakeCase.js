export const camelToSnakeCase = str => str.replace(/([A-Z])/g, group => `_${group.toLowerCase()}`)
