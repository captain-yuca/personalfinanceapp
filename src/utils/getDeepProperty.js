export const getDeepProperty = (path, obj) =>
  path.split('.').reduce((prev, curr) => (prev ? prev[curr] : null), obj || null)
