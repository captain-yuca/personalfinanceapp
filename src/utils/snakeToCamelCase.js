export const snakeToCamelCase = str =>
  str.replace(/([-_][a-z])/g, group => group.toUpperCase().replace('-', '').replace('_', ''))
