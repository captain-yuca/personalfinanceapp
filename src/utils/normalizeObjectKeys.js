import { camelToSnakeCase } from './camelToSnakeCase'
import { snakeToCamelCase } from './snakeToCamelCase'

export const normalizeObjectKeys = (obj, toSnake = false) =>
  Object.entries(obj).reduce((accum, [rawKey, value]) => {
    const key = toSnake ? camelToSnakeCase(rawKey) : snakeToCamelCase(rawKey)
    return { ...accum, [key]: value }
  }, {})
