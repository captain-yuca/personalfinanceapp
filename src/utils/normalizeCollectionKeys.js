import { normalizeObjectKeys } from './normalizeObjectKeys'

export const normalizeCollectionKeys = (collection, toSnake = false) =>
  collection.map(item => normalizeObjectKeys(item, toSnake))
