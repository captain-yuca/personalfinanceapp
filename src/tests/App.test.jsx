import React from 'react'
import { render } from '../test.utils'

import App from '../App'

describe('App component', () => {
  it('should render correctly', () => {
    const { container } = render(<App />)
    expect(container).toMatchSnapshot()
  })
})
