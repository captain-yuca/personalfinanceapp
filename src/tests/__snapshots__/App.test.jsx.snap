// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component should render correctly 1`] = `
.c2 {
  position: absolute;
  left: 50%;
  top: 100px;
  -webkit-transform: translateX(-50%);
  -ms-transform: translateX(-50%);
  transform: translateX(-50%);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c3 {
  width: 170px;
}

.c4 {
  font-size: 36px;
  margin: 16px 0;
}

.c5 {
  color: #61dafb;
  font-weight: bold;
  margin: 0 5px;
  display: inline;
}

.c6 {
  color: #ff0000;
  font-weight: bold;
  margin: 0 5px;
  display: inline;
}

.c0 {
  width: 100vw;
  height: 100vh;
  overflow: hidden;
  position: relative;
  background-color: #212121;
  color: #ffffff;
}

.c1 {
  position: absolute;
  width: 130px;
  height: 130px;
  right: 20px;
  bottom: 20px;
  background-image: url(../assets/images/react.svg);
  background-size: contain;
  background-repeat: no-repeat;
  background-position: center;
  -webkit-animation: fvtopB 2.5s linear infinite;
  animation: fvtopB 2.5s linear infinite;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    />
    <div
      class="c2"
    >
      <img
        alt="MAD DEVS"
        class="c3"
        src="md.svg"
      />
      <h2
        class="c4"
      >
        React MAD Boiler
      </h2>
      <p
        class=""
      >
        Simple boilerplate for
        <a
          class="c5"
          color="react"
          href="https://ru.reactjs.org/"
          target="_blank"
        >
          React
        </a>
        from
        <a
          class="c6"
          color="primary"
          href="https://maddevs.io/"
          target="_blank"
        >
          Mad Devs
        </a>
      </p>
    </div>
  </div>
</div>
`;
